### PR TITLE
Reorganize Screening Page Layout

### PR DESCRIPTION
**Pull request by casey1173**
Updates the literature reference screening/tagging page to account for user feedback and conflict resolution updates.
 - Wording, text size, and button placements changed for clarity and workflow optimization
 - "Tagged" and "Untagged" reference lists merged into one list, with an icon indicating tagged references
 - "Save and go to next untagged" button is now "Save and go to next"; loops around to start of list once complete
 - Tag color changed from maroon to a navy blue
 - Screening instructions added as a modal, and as a toggleable div below abstract
 - Abstract div element is now scrollable so that tags are kept on screen while reading long abstracts
 - Added a toggle to show full tag depth or just the leaf node on tags; full tag is shown while hovering
 - Moved "click to remove" help text from a tag popover to its own help text popover
 - Made reference list collapsible

Before:
image

After:
image

New actions menu
image

Screening instructions modal
image

Screening instructions pinned to page
image

Reference list collapsed
image

### CODE REVIEW

File: frontend/lit/TagReferences/Main.js [hunk 1]
```diff
@@ -114,55 +120,46 @@ class TagReferencesMain extends Component {
...
                         tagtree={toJS(store.tagtree)}
                         handleTagClick={tag => store.addTag(tag)}
                     />
+                    <div className="alert alert-info mt-3">
+                        <b>Screening Instructions:</b>
+                        <button type="button" className="close" data-dismiss="alert">
+                            &times;
+                        </button>
+                        {store.config.instructions}
+                    </div>
```
Comment (casey1173) [inline frontend/lit/TagReferences/Main.js:162]: @shapiromatron what's the best way to clean the quill field input? 
image

Comment (shapiromatron) [reply frontend/lit/TagReferences/Main.js:162]: you dont :) https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml

We know the data are "safe" b/c we clean from the user

File: frontend/lit/TagReferences/Main.js [hunk 2]
```diff
@@ -186,7 +189,13 @@ class TagReferencesMain extends Component {
                                 }}>
                                 &times;
                             </button>
-                            <div dangerouslySetInnerHTML={{__html: store.config.instructions}} />
+                            <div
+                                dangerouslySetInnerHTML={
+                                    store.config.instructions.length > 0
+                                        ? {__html: store.config.instructions}
+                                        : {__html: "No screening instructions found."}
```
Comment (shapiromatron) [inline frontend/lit/TagReferences/Main.js:196]: can we just hide the screening instructions button/div instead if none are available?

Comment (casey1173) [reply frontend/lit/TagReferences/Main.js]: good idea!! didn't think about hiding the button...

File: hawc/apps/lit/views.py [hunk 3]
```diff
@@ -264,12 +264,14 @@ def get_app_config(self, context) -> WebappConfig:
         if hasattr(self, "qs_reference"):
             refs = self.qs_reference
         else:
-            refs = refs = models.Reference.objects.filter(**self.get_ref_qs_filters()).distinct()
-        refs = refs.prefetch_related("searches", "identifiers", "tags")
+            refs = models.Reference.objects.filter(**self.get_ref_qs_filters()).distinct()
+        refs = refs.select_related("study").prefetch_related("searches", "identifiers", "tags")
```
Comment (shapiromatron) [inline hawc/apps/lit/views.py:268]: fixed an n+1 query issue

File: frontend/lit/components/Reference.js [hunk 4]
```diff
@@ -9,6 +9,33 @@ import Hero from "shared/utils/Hero";
 
 import ReferenceButton from "./ReferenceButton";
 
+const markKeywords = (text, allSettings) => {
+        const all_tokens = allSettings.set1.keywords
+                .concat(allSettings.set2.keywords)
+                .concat(allSettings.set3.keywords),
+            all_re = new RegExp(all_tokens.join("|"), "gim");
+
+        if (all_tokens.length === 0) {
```
Comment (shapiromatron) [inline frontend/lit/components/Reference.js:18]: found an edge case too if all tokens are null it highlights everything; so i added a few guards to bail early if that's the case...

**Review by shapiromatron — APPROVED** [review]
Looks fantastic Casey I think people will really like the changes. I made a number of small changes, and lots of commits. Please take a look through them and if you're ok with changes, ready to merge for me!

**Pull request merged**